### PR TITLE
feat(within): Support cy.within

### PR DESCRIPTION
**Pull request by npeterkamps**
**What / How**:

* Support `cy.within` by using `cy.state('withinSubject')` when available.
* Support passing in a container within the options object.
* Since Cypress uses jQuery elements and `dom-testing-library` expects DOM nodes, created a helper to retrieve the DOM node. 
  * Used [isJquery provided by Cypress](https://github.com/cypress-io/cypress/blob/master/packages/driver/src/dom/jquery.coffee), since `instanceof` didn't work.
* Installed wait-port and use it within `test:cypress:run` to avoid race-conditions.

**Why**:

See #8 

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Ready to be merged 
* [x] Added myself to contributors table

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {queries, waitForElement} from 'dom-testing-library'
+import {getContainer} from './support/utils'
```
[the remark this answers is not in the document]
Comment (npeterkamps) [reply src/index.js]: Well, I remembered you already had a utils file within your code from #6:
```
import {
  makeCypressCommandLog,
  makeElementDebugString,
  getNodeFullText,
} from '../support/utils';
```

So I just tried to match that 😉

[the remark this answers is not in the document]
Comment (sompylasar) [reply src/index.js]: The snippet I have shown is in the `cypress` folder of my app where Cypress defines the file structure, it's not in the library code.

**Comment by sompylasar** [comment]
@npeterkamps Thanks for the fixes in https://github.com/kentcdodds/cypress-testing-library/pull/11/commits/6394e7e3b24485b2dda95cef4e3e5e4e9a9cee68 – I apologize for the confusion that I brought, but I did not notice that `command` takes `cy` as its argument, propagated from `add-commands`:

https://github.com/kentcdodds/cypress-testing-library/blob/7d14b896f6f68775e962d03928a9023f325e0791/src/add-commands.js#L1-L7

I wonder why `Cypress` is still accessed from the globals, but `cy` is made injected as a local. Consistency of where to take the references to these objects from is what I try to ensure. Either take both `Cypress` and `cy` from globals everywhere, or take both from globals in `add-commands`, but otherwise take them as local variables from `command` arguments and propagate to any utility function that needs them.

**Comment by npeterkamps** [comment]
> Is container currently the only option across all the library APIs where the library takes a DOM element or jQuery collection?

Yes. The only API that `cypress-testing-library` exposes is the command function that is created for each of the query functions. Looking at the documentation from `dom-testing-library`, there are two `HTMLElement` parameters; the one passed to the query functions and the one within the options object for `waitForElement`. So as far as I'm aware, we're supporting all current options.

> I'm questioning this quite broad statement that "cypress-testing-library supports both jQuery elements and DOM nodes." – if we want it to be so, we have to ensure this is true for all the current and future APIs.

Since Cypress uses jQuery elements and `dom-testing-library` expects DOM nodes, I think it would be quite inconvenient if we don't automatically convert jQuery input to DOM nodes. So I think we do need to ensure that we support both (in the future as well).

**Comment by sompylasar** [comment]
Thanks @npeterkamps, all looks good.

**Comment by kentcdodds** [comment]
Thanks friends!

**Comment by sompylasar** [comment]
@kentcdodds I didn't find this in `CONTRIBUTING.md`, which option should I use just in case I need to merge a PR here (squash and merge? on desktop only? how to compose the message?).

**Comment by kentcdodds** [comment]
I think it's in the maintaining.md in the `other` directory. It's squash and merge and you need to make sure that the commit message is right and you can do that in the GitHub UI

**Comment by sompylasar** [comment]
@kentcdodds Found it, thanks!

**Pull request merged**